### PR TITLE
Add typing for steps/prepare/shell.py

### PR DESCRIPTION
**Pull request by guy9050**

### CODE REVIEW

File: tmt/steps/prepare/shell.py [hunk 1]
```diff
@@ -30,28 +33,28 @@ class PrepareShell(tmt.steps.prepare.PreparePlugin):
...
-    def wake(self, keys=None):
+    def wake(self, keys: Optional[List[str]] = None) -> None:
         """ Wake up the plugin, process data, apply options """
         super().wake(keys=keys)
 
         # Convert to list if single script provided
         tmt.utils.listify(self.data, keys=['script'])
 
-    def go(self, guest):
+    def go(self, guest: 'Guest') -> None:
```
Comment (happz) [inline tmt/steps/prepare/shell.py:57]: You don't need string wrap when a class can be imported in runtime, which si the case here. No special care needed.

File: tmt/steps/prepare/shell.py [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 import click
 import fmf
 
 import tmt
 import tmt.utils
+from tmt.steps.provision import Guest
 
 
-class PrepareShell(tmt.steps.prepare.PreparePlugin):
+class PrepareShell(tmt.steps.prepare.PreparePlugin):  # type: ignore[misc]
```
Comment (thrix) [inline tmt/steps/prepare/shell.py:11]: is this ignore something we want to remove later right. Should we mark this as `TODO` item? or we will create a task to go through this once we are done @happz ?

Comment (happz) [reply tmt/steps/prepare/shell.py:11]: Yes, add a TODO, please. There is no issue yet, I'll add a note or two to our type annotation issue.

Comment (thrix) [reply tmt/steps/prepare/shell.py:11]:
```suggestion
# TODO: drop ignore once type annotations between modules enabled
class PrepareShell(tmt.steps.prepare.PreparePlugin):  # type: ignore[misc]
```

**Review by psss — APPROVED** [review]
Looks good, just converted `how` into `str`. Rebased on #1193 so that we can merge at once when tests are green.

**Pull request merged**